src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509CertificateReader.cs

```diff
@@ -298,10 +332,25 @@ public string GetNameInfo(X509NameType nameType, bool forIssuer)
 
         public void AppendPrivateKeyInfo(StringBuilder sb)
         {
+            if (!HasPrivateKey)
+            {
+                return;
+            }
+
+            // There's nothing really to say about the key, just acknowledge there is one.
+            sb.AppendLine();
+            sb.AppendLine();
+            sb.AppendLine("[Private Key]");
```

Comment:
Apparently not: http://referencesource.microsoft.com/#System/security/system/security/cryptography/x509/x509certificate2.cs,1229

But don't look at that code too long, or you'll be upset at their lack of using AppendLine and instead to make a lot of intermediate temporary strings :smile:.